core/appeal/service.go

```diff
@@ -289,6 +293,21 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
 						},
 					},
 				})
+
+				if appeal.AccountType == domain.DefaultAppealAccountType && appeal.AccountID != appeal.CreatedBy {
```

Comment:
Can we add a comment explaining this? Or is it possible to extract this out as a function? I saw we are doing similar thing in another place.

Reply:
+1 Let's have both - separating as a function and then comments there to detail why.

Reply:
done